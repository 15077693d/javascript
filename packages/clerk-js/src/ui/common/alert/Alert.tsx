import React from 'react';

export type AlertType = 'info' | 'error';

export type AlertProps = {
  type: AlertType;
  children: React.ReactNode;
  style?: {};
} & React.HTMLAttributes<HTMLDivElement>;

// Renders global alerts across components, will be replaced by notification snackbars.
export function Alert({ type, children, style }: AlertProps): JSX.Element {
  if (!children) {
    return <></>;
  }

  return (
    <div
      className={`cl-${type}`}
      style={style}
    >
      {children}
    </div>
  );
}
