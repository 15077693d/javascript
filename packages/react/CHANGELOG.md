# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

### [2.11.2-staging.0](https://github.com/clerkinc/javascript/compare/@clerk/clerk-react@2.11.1...@clerk/clerk-react@2.11.2-staging.0) (2022-02-15)

**Note:** Version bump only for package @clerk/clerk-react





### [2.11.1](https://github.com/clerkinc/javascript/compare/@clerk/clerk-react@2.11.1-staging.0...@clerk/clerk-react@2.11.1) (2022-02-14)

**Note:** Version bump only for package @clerk/clerk-react





### 2.11.1-staging.0 (2022-02-11)


### Features

* **clerk-sdk-node:** Deprecate Session named middleware, introduce withAuth, requireAuth ([4e69553](https://github.com/clerkinc/javascript/commit/4e695535e41fe7c135cbf303a0d021e7b7d30f7d))
