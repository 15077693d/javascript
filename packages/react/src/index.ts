export * from './contexts';
export * from './components';
export type { ClerkProp } from './types';
export { isMagicLinkError, MagicLinkErrorCode } from './errors';
export * from './hooks';
