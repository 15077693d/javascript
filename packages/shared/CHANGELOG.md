# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

### [0.0.16-alpha.4](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.3...@clerk/shared@0.0.16-alpha.4) (2022-03-23)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.3](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.2...@clerk/shared@0.0.16-alpha.3) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.2) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.1) (2022-03-20)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.0) (2022-03-19)

**Note:** Version bump only for package @clerk/shared

### [0.0.15](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.1...@clerk/shared@0.0.15) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.15-staging.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.0...@clerk/shared@0.0.15-staging.1) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.13](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12...@clerk/shared@0.0.13) (2022-03-11)

**Note:** Version bump only for package @clerk/shared

### [0.0.12](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12-staging.0...@clerk/shared@0.0.12) (2022-03-09)

**Note:** Version bump only for package @clerk/shared

### [0.0.10](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.9...@clerk/shared@0.0.10) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.9](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.8...@clerk/shared@0.0.9) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.8](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.7...@clerk/shared@0.0.8) (2022-03-03)

**Note:** Version bump only for package @clerk/shared

### [0.0.7](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.6...@clerk/shared@0.0.7) (2022-03-02)

**Note:** Version bump only for package @clerk/shared

### [0.0.6](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5...@clerk/shared@0.0.6) (2022-03-01)

### Bug Fixes

- **shared:** Phone input should start blank ([43f0b06](https://github.com/clerkinc/clerk_docker/commit/43f0b0603608866f6b9e0a37a284c0ea72c0004b))

### [0.0.5](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5-staging.0...@clerk/shared@0.0.5) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.5-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.4-staging.0...@clerk/shared@0.0.5-staging.0) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.4-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.3-staging.0...@clerk/shared@0.0.4-staging.0) (2022-02-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.3-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2...@clerk/shared@0.0.3-staging.0) (2022-02-15)

**Note:** Version bump only for package @clerk/shared

### [0.0.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2-staging.0...@clerk/shared@0.0.2) (2022-02-14)

**Note:** Version bump only for package @clerk/shared

### 0.0.2-staging.0 (2022-02-11)

**Note:** Version bump only for package @clerk/shared
