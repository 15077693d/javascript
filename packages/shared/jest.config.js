/** @type {import('ts-jest').JestConfigWithTsJest} */
const config = {
  displayName: 'shared',
  injectGlobals: true,

  testEnvironment: 'jsdom',
  roots: ['<rootDir>/src'],
  setupFiles: ['./setupJest.ts'],
  testRegex: ['/ui/.*/__tests__/.*.test.[jt]sx?$', '/(core|utils)/.*.test.[jt]sx?$'],
  testPathIgnorePatterns: ['/node_modules/'],

  collectCoverage: false,
  coverageProvider: 'v8',
  coverageDirectory: 'coverage',
  coveragePathIgnorePatterns: ['/node_modules/'],

  moduleDirectories: ['node_modules', '<rootDir>/src'],
  transform: {
    '^.+\\.m?tsx?$': ['ts-jest', { tsconfig: 'tsconfig.test.json', diagnostics: false }],
  },
};

module.exports = config;
