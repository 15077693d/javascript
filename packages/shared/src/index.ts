/** The following files are not exported on purpose:
 * - cookie.ts
 * - globs.ts
 *
 * The following folders are also not exported on purpose:
 * - error
 * - react
 * - utils
 *
 * People should always use @clerk/shared/<name> instead
 */

export { createWorkerTimers } from './workerTimers';
export * from './browser';
export * from './color';
export * from './date';
export * from './deprecated';
export * from './file';
export { isomorphicAtob } from './isomorphicAtob';
export * from './keys';
export { loadScript } from './loadScript';
export { LocalStorageBroadcastChannel } from './localStorageBroadcastChannel';
export * from './poller';
export * from './proxy';
export * from './underscore';
export * from './url';
