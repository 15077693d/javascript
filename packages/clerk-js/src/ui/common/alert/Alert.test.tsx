import { renderJSON } from '@clerk/shared/testUtils';
import * as React from 'react';

import { Alert } from './Alert';

describe('alert for info', () => {
  it('renders an info alert', () => {
    const tree = renderJSON(<Alert type='error'>Foo</Alert>);
    expect(tree).toMatchSnapshot();
  });
});

describe('alert for error', () => {
  it('renders an error alert', () => {
    const tree = renderJSON(<Alert type='info'>Bar</Alert>);
    expect(tree).toMatchSnapshot();
  });
});
