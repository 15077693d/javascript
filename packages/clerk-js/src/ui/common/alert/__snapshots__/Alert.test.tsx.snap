// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert for error renders an error alert 1`] = `
<div
  className="cl-info"
>
  Bar
</div>
`;

exports[`alert for info renders an info alert 1`] = `
<div
  className="cl-error"
>
  Foo
</div>
`;
